docs/messaging/rabbitmq.md

```diff
@@ -14,13 +14,19 @@ The Semantic Conventions for [RabbitMQ](https://www.rabbitmq.com/) extend and ov
 
 ### Destination name
 
-In RabbitMQ, the destination is defined by an *exchange* and a *routing key*.
-`messaging.destination.name` MUST be set according to the following rules:
+In RabbitMQ, the destination is defined by an *exchange*, a *routing key* and
+for consumers, a *queue*.
+`messaging.destination.name` MUST be set to:
 
-- Both *exchange* and *routing key* present and non-empty: `{exchange}:{routing key}`
-- Only *exchange* present and non-empty: `{exchange}`
-- Only *routing key* present and non-empty: `{routing key}`
-- Otherwise: `amp.default`
+**Producers**
+
+- `{exchange}:{routing key}` when both values are present and non-empty.
+If any has an empty value (e.g., the default exchange is used) it SHOULD be omitted.
+- Otherwise: `amq.default`
+
+**Consumers**
+- `{exchange}:{routing key}:{queue}` when all values are present and non-empty.
```

Comment:
maybe check if routing key and queue are the same and populate just one

Reply:
Yeah I looked and it seems when receiving/delivering, only the routing key is present https://www.rabbitmq.com/docs/consumers#message-properties. Also, if I understand it right from some research, the routing key can be then also the same as queue name https://stackoverflow.com/questions/22655735/retrieve-name-of-rabbitmq-queue-where-message-was-consumed-from

So I think we should be good to keep only the exchange and routing key. PTAL

Reply:
wait, the queue is available on the client level -https://www.rabbitmq.com/client-libraries/java-api-guide#consuming

```java
boolean autoAck = false;
channel.basicConsume(queueName, autoAck, "myConsumerTag",
     new DefaultConsumer(channel) {
         @Override
         public void handleDelivery(String consumerTag,
                                    Envelope envelope,
                                    AMQP.BasicProperties properties,
                                    byte[] body)
             throws IOException
         {
             String routingKey = envelope.getRoutingKey();
             String contentType = properties.getContentType();
             long deliveryTag = envelope.getDeliveryTag();
             <i>// (process the message components here ...)</i>
             channel.basicAck(deliveryTag, false);
         }
     });
```

and most of the instrumentations should be able to populate it. We need to capture it somewhere

Reply:
Yes, I checked and it's correct. I added it back, only now stating when both routing key and queue name are the same, only one should be used.